src/DynamoCore/Core/PathManager.cs

```diff
@@ -304,18 +314,33 @@ internal PathManager(PathManagerParams pathManagerParams)
         /// is referring to. This method call throws exception if any creation 
         /// fails.
         /// </summary>
-        internal void EnsureDirectoryExistence()
+        /// <param name="exceptions">The output list of exception, if any of 
+        /// the target directories cannot be created during this call.</param>
+        internal void EnsureDirectoryExistence(List<Exception> exceptions)
         {
+            if (rootDirectories.Count <= 0)
+            {
+                throw new InvalidOperationException(
+                    "At least one custom package directory must be specified");
+            }
+
+            if (exceptions == null)
+                throw new ArgumentNullException("exceptions");
+
+            exceptions.Clear();
+
             // User specific data folders.
-            CreateFolderIfNotExist(userDataDir);
-            CreateFolderIfNotExist(userDefinitions);
-            CreateFolderIfNotExist(logDirectory);
-            CreateFolderIfNotExist(packagesDirectory);
-            CreateFolderIfNotExist(backupDirectory);
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(userDataDir));
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(DefaultUserDefinitions));
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(logDirectory));
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(DefaultPackagesDirectory));
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(backupDirectory));
 
             // Common data folders for all users.
-            CreateFolderIfNotExist(commonDataDir);
-            CreateFolderIfNotExist(commonDefinitions);
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(commonDataDir));
+            exceptions.Add(PathHelper.CreateFolderIfNotExist(commonDefinitions));
+
+            exceptions.RemoveAll(x => x == null); // Remove all null entries.
         }
```

Comment:
Hi Appu, I have tried to log error message from within `EnsureDirectoryExistence` method, but that was done before `Logger` is created. The `Logger` needs `PathManager.LogDirectory` property, with the order of creation, you can see that we can't move `Logger` up before `EnsureDirectoryExistence` method is called:

```csharp
protected DynamoModel(IStartConfiguration config)
{
    // ...

    // This is where PathManager is created.
    pathManager = new PathManager(new PathManagerParams
    {
        CorePath = config.DynamoCorePath,
        PathResolver = config.PathResolver
    });

    // This is where the exception is thrown, and it is the place where 
    // "LogDirectory" is created.
    pathManager.EnsureDirectoryExistence();

    // This is where Logger is created, but it needs "LogDirectory" to 
    // exist at this point.
    Logger = new DynamoLogger(DebugSettings, pathManager.LogDirectory);

    // ...
}
```

Since that's the case, I have updated the code to return a list of `Exception` objects from `EnsureDirectoryExistence` method, then after `Logger` is created, it logs these `Exception` objects, if any.